modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js

```diff
@@ -377,6 +364,33 @@ function transformVertexArray(
   return newVertices;
 }
 
+function transformVertexPositions(vertexVector, calleeArgs) {
+  const {cartesianModelMatrix, cartographicOrigin, nodeMatrix} = calleeArgs;
+
+  if (nodeMatrix) {
+    vertexVector = vertexVector.transform(nodeMatrix);
+  }
+
+  vertexVector = vertexVector.transform(cartesianModelMatrix);
+  Ellipsoid.WGS84.cartesianToCartographic(
+    [vertexVector[0], vertexVector[1], vertexVector[2]],
+    vertexVector
+  );
+  vertexVector = vertexVector.subtract(cartographicOrigin);
+  return vertexVector;
+}
+
+function transformVertexNormals(vertexVector, calleeArgs) {
+  const {cartesianModelMatrix, nodeMatrix} = calleeArgs;
+
+  if (nodeMatrix) {
```

Comment:
I think we need to move into cartographic coordinates here as well. the normals need to be in the same coordinate system, just without translation.

Reply:
I will check, but in my understanding `cartesianModelMatrix` applying exactly for this purpose.